### PR TITLE
kubectl/run: remove log "object created" when --dry-run=true

### PR DESCRIPTION
**Pull request by hodovska**
Since no object is created when --dry-run=true there is no point in showing log: "object created".

bug 1270697
Bugzilla link https://bugzilla.redhat.com/show_bug.cgi?id=1270697

This is the output on my instance:

```
$ kubectl run example --image=busybox --dry-run=true
NAME       DESIRED   CURRENT   UP-TO-DATE   AVAILABLE   AGE
example   1         0         0            0           <unknown>
```

### CODE REVIEW

**Comment by deads2k** [comment]
What output do I get now?

**Comment by hodovska** [comment]
When not specified, the output is similar to output of get command.

**Comment by deads2k** [comment]
> When not specified, the output is similar to output of get command.

Can you post an example?  It's not obvious to me that hitting this case: https://github.com/kubernetes/kubernetes/blob/master/pkg/kubectl/resource_printer.go#L136 will do what I want.

**Comment by hodovska** [comment]
I think this is the place where the proper printer is set when the outputFormat=="" https://github.com/kubernetes/kubernetes/blob/master/pkg/kubectl/cmd/util/factory.go#L1224

**Comment by deads2k** [comment]
> I think this is the place where the proper printer is set when the outputFormat=="" https://github.com/kubernetes/kubernetes/blob/master/pkg/kubectl/cmd/util/factory.go#L1224

ok, how about that output?

**Comment by hodovska** [comment]
This is the output on my instance:
```
$ k run example --image=busybox --dry-run=true
NAME       DESIRED   CURRENT   UP-TO-DATE   AVAILABLE   AGE
exampple   1         0         0            0           <unknown>
```

**Comment by ncdc** [comment]
@hodovska could you please add a test case to `hack/test-cmd.sh`? Thanks!

**Comment by deads2k** [comment]
Code looks fine, but its not the output I'd expect.

@fabianofranz for UI.

**Comment by hodovska** [comment]
If it matched the output of get cmd without the flag it would print only new line, but this is not what is specified in --help:

```
--dry-run[=false]: If true, only print the object that would be sent, without sending it.
```

Possibilities are:
1, Choose one kind of output (e.g. yaml) which would be default for --dry-run
2, Create default output teplate, where would be all the possible values printed - the ones which can be affected by run cmd flags
3, Update --help and hint user to use also the output flag when --dry-run is used:
```
--dry-run[=false]: If true, the object is not actually created. Use -o flag to see object information.
```
4, Whatever other suggestion you have

File: hack/lib/test.sh [hunk 1]
```diff
@@ -223,3 +223,38 @@ kube::test::if_has_string() {
...
+    echo ${reset}${red}
+    caller
+    echo ${reset}
+    return 1
+  fi
+
+  if [[ $(echo "$out" | grep "created") ]]; then
+    echo ${bold}${red}
+    echo "FAIL!"
+    echo "run with --dry-run-true"
```
Comment (soltysh) [inline hack/lib/test.sh]: Typo: `--dry-run=true`

**Comment by fabianofranz** [comment]
I don't think it's expected from the output of `--dry-run` anything other than what you would get for an output without it. I'd say the output of the regular use case (`run` without `--dry-run`) clearly has to be improved to print what's being created instead of just the message, and then we'd have the same output in case of `--dry-run`.

**Comment by hodovska** [comment]
Duplicate of #28606

**Pull request closed without merging**